src/dbnode/persist/fs/migration/migration_test.go

```diff
@@ -0,0 +1,154 @@
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/digest"
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/persist/fs/msgpack"
+	"github.com/m3db/m3/src/dbnode/storage"
+	"github.com/m3db/m3/src/x/checked"
+	"github.com/m3db/m3/src/x/ident"
+	"github.com/stretchr/testify/require"
+)
+
+func TestToVersion1_1Run(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	var shard uint32 = 1
+	nsId := ident.StringID("foo")
+
+	// Write unmigrated fileset to disk
+	fsOpts := writeUnmigratedData(t, filePathPrefix, nsId, shard)
+
+	// Read info file of just written fileset
+	infoFileResult := fs.ReadInfoFiles(filePathPrefix, nsId, shard,
+		fsOpts.InfoReaderBufferSize(), fsOpts.DecodingOptions())[0]
+	indexFd, err := os.Open(path.Join(fsOpts.FilePathPrefix(), fmt.Sprintf("data/%s/%d/fileset-%d-0-index.db",
+		nsId.String(), shard, infoFileResult.Info.BlockStart)))
+	require.NoError(t, err)
+	oldBytes, err := ioutil.ReadAll(indexFd)
+	require.NoError(t, err)
+
+	// Configure and run migration
+	pm, err := fs.NewPersistManager(fsOpts.SetEncoder(msgpack.NewEncoder())) // Set encoder to most up-to-date version
+	require.NoError(t, err)
+
+	md, err := namespace.NewMetadata(nsId, namespace.NewOptions())
+	require.NoError(t, err)
+
+	opts := NewTaskOptions().
+		SetNewMergerFn(fs.NewMerger).
+		SetPersistManager(pm).
+		SetNamespaceMetadata(md).
+		SetStorageOptions(storage.NewOptions()).
+		SetShard(shard).
+		SetInfoFileResult(infoFileResult).
+		SetFilesystemOptions(fsOpts)
+
+	task, err := NewToVersion1_1Task(opts)
+	require.NoError(t, err)
+
+	err = task.Run()
+	require.NoError(t, err)
+
+	// Read the index entries of new volume set
+	indexFd, err = os.Open(path.Join(fsOpts.FilePathPrefix(), fmt.Sprintf("data/%s/%d/fileset-%d-1-index.db",
+		nsId.String(), shard, infoFileResult.Info.BlockStart)))
+	require.NoError(t, err)
+	newBytes, err := ioutil.ReadAll(indexFd)
+	require.NoError(t, err)
+
+	// Diff bytes of unmigrated vs migrated fileset
+	require.NotEqual(t, oldBytes, newBytes)
+
+	// Corrupt bytes to trip newly added checksum
+	decoder := msgpack.NewDecoder(nil)
+	newBytes[len(newBytes)-1] = 'x'
+	decoder.Reset(msgpack.NewByteDecoderStream(newBytes))
+	_, err = decoder.DecodeIndexEntry(nil)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "checksum mismatch")
+}
+
+func writeUnmigratedData(t *testing.T, filePathPrefix string, nsId ident.ID, shard uint32) fs.Options {
+	// Use an encoder that doesn't generate entry level checksums
+	eOpts := msgpack.LegacyEncodingOptions{EncodeLegacyIndexEntryVersion: msgpack.LegacyEncodingIndexEntryVersionV2}
+	enc := msgpack.NewEncoderWithOptions(eOpts)
+
+	// Write data
+	fsOpts := fs.NewOptions().
+		SetFilePathPrefix(filePathPrefix).
+		SetEncoder(enc)
+	w, err := fs.NewWriter(fsOpts)
+	require.NoError(t, err)
+
+	blockStart := time.Now()
```

Comment:
nit: may be worth adding `.Truncate(time.Hour)` to tests in general just to make output easier to read